maat-court-data-api/src/main/java/gov/uk/courtdata/dces/service/FdcContributionsTestService.java

```diff
@@ -0,0 +1,115 @@
+package gov.uk.courtdata.dces.service;
+
+import gov.uk.courtdata.dces.request.CreateFdcTestDataRequest;
+import gov.uk.courtdata.dces.request.FdcNegativeTestType;
+import gov.uk.courtdata.entity.FdcContributionsEntity;
+import gov.uk.courtdata.entity.FdcItemsEntity;
+import gov.uk.courtdata.entity.RepOrderEntity;
+import gov.uk.courtdata.enums.FdcContributionsStatus;
+import gov.uk.courtdata.repository.CrownCourtProcessingRepository;
+import gov.uk.courtdata.repository.FdcContributionsRepository;
+import gov.uk.courtdata.repository.FdcItemsRepository;
+import gov.uk.courtdata.repository.RepOrderRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+@ConditionalOnProperty(prefix = "sentry", name = "environment", havingValue = "development")
+public class FdcContributionsTestService {
+    private final FdcItemsRepository fdcItemsRepository;
+    private final RepOrderRepository repOrderRepository;
+    private final CrownCourtProcessingRepository repOrderCrownCourtOutcomeRepository;
+    private final FdcContributionsRepository fdcContributionsRepository;
+    private final DebtCollectionRepository debtCollectionRepository;
+
+
+    @Transactional
+    public List<Integer> createFdcMergeTestData(CreateFdcTestDataRequest request){
+        boolean isNegativeTest = request.isNegativeTest();
+        FdcNegativeTestType negativeTestType = isNegativeTest ? request.getNegativeTestType() : null; // only populate if negative is wanted.
+        int numberOfTestEntries = request.getNumOfTestEntries();
+
+        // find candidates
+        List<Integer> repOrderIds = debtCollectionRepository.getMerge1TestCandidates(numberOfTestEntries);
+        if(FdcNegativeTestType.SOD.equals(negativeTestType)){
+            LocalDate targetDate = LocalDate.now().plus(3, ChronoUnit.MONTHS);
+            setSentenceOrderDate(repOrderIds, targetDate);
+        }
+        if(FdcNegativeTestType.CCO.equals(negativeTestType)){
```

Comment:
There are a lot of conditions going on in this methods with comments - I'd suggest pulling each condition block in to its own method and then the comments would not be required.